src/App.tsx

```diff
@@ -1,29 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
+// import { event } from 'cypress/types/jquery';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const DEFAULT_PAGE = 1;
+const items: string[] = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(DEFAULT_PAGE);
+
+  const startItem = (currentPage - 1) * itemsPerPage;
+  const endItem =
+    startItem + itemsPerPage > items.length
+      ? items.length
+      : startItem + itemsPerPage;
```

Comment:
The logic for determining the `endItem` is repeated twice (`startItem + itemsPerPage`). Consider storing the result of this calculation in a variable to avoid repetition and improve readability.